modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/DXPActivationKeysTable/index.js

```diff
@@ -0,0 +1,239 @@
+	const [statusBar, setStatusBar] = useState({});
+
+	const [activationKeysFiltered, setActivationKeysFiltered] = useState([]);
+	const [totalCount, setTotalCount] = useState(5);
+
+	const [filterStatusBar, setFilterStatusBar] = useState('all');
+
+	const [activePage, setActivePage] = useState(1);
+	const [itemsPerPage, setItemsPerPage] = useState(5);
+	const [isLoadingActivationKeys, setIsLoadingActivationKeys] = useState(
+		false
+	);
+
+	useEffect(() => {
+		if (filterStatusBar) {
+			setActivePage(1);
+		}
+	}, [filterStatusBar]);
+
+	useEffect(() => {
+		setIsLoadingActivationKeys(true);
+		const fetchActivationKeysData = async () => {
+			const {items} = await getActivationLicenseKey(
+				project.accountKey,
+				licenseKeyDownloadURL,
+				encodeURI('active eq true'),
+				PAGE,
+				MAX_ITEMS,
+				sessionId
+			);
+			if (items) {
+				setActivationKeys(items);
+				setStatusBar({
+					activeTotalCount: items.filter((activationKey) =>
+						ACTIVATION_KEYS_LICENSE_FILTER_TYPES.active(
+							activationKey
+						)
+					).length,
+					allTotalCount: items.length,
+					expiredTotalCount: items.filter((activationKey) =>
+						ACTIVATION_KEYS_LICENSE_FILTER_TYPES.expired(
+							activationKey
+						)
+					).length,
+					notActiveTotalCount: items.filter((activationKey) =>
+						ACTIVATION_KEYS_LICENSE_FILTER_TYPES.notActivated(
+							activationKey
+						)
+					).length,
+				});
+			}
+
+			setIsLoadingActivationKeys(false);
+		};
+		fetchActivationKeysData();
+	}, [licenseKeyDownloadURL, project, sessionId]);
+
+	useEffect(() => {
+		if (activationKeys.length) {
+			const activationKeysFilterData = activationKeys.filter(
+				(activationKey) =>
+					ACTIVATION_KEYS_LICENSE_FILTER_TYPES[filterStatusBar]
+						? ACTIVATION_KEYS_LICENSE_FILTER_TYPES[filterStatusBar](
+								activationKey
+						  )
+						: Boolean
+			);
+			setTotalCount(activationKeysFilterData.length || 5);
+			const activationKeysFilterByPage = activationKeysFilterData.slice(
+				itemsPerPage * activePage - itemsPerPage,
+				itemsPerPage * activePage
+			);
+			setActivationKeysFiltered(
+				activationKeysFilterByPage.length
+					? activationKeysFilterByPage
+					: activationKeysFilterData
+			);
+		}
+	}, [activationKeys, activePage, filterStatusBar, itemsPerPage]);
+
+	return (
+		<div>
+			<div className="align-center cp-dxp-activation-key-container d-flex justify-content-between">
+				<h3 className="m-0">Activation Keys</h3>
+
+				<RoundedGroupButtons
+					groupButtons={[
+						getGroupButtons(
+							ACTIVATION_STATUS.all,
+							statusBar?.allTotalCount
+						),
+						getGroupButtons(
+							ACTIVATION_STATUS.active,
+							statusBar?.activeTotalCount
+						),
+						getGroupButtons(
+							ACTIVATION_STATUS.notActivated,
+							statusBar?.notActiveTotalCount
+						),
+						getGroupButtons(
+							ACTIVATION_STATUS.expired,
+							statusBar?.expiredTotalCount
+						),
+					]}
```

Comment:
Is it possible to move the logic for a variable?

```suggestion
					groupButtons={groupButtons}
```